src/jit/optimizer.cpp

```diff
@@ -993,8 +993,7 @@ bool Compiler::optIsLoopTestEvalIntoTemp(GenTreePtr testStmt, GenTreePtr* newTes
     // Make sure we have jtrue (vtmp != 0)
     if ((relop->OperGet() == GT_NE) && 
         (opr1->OperGet() == GT_LCL_VAR) &&
-        (opr2->OperGet() == GT_CNS_INT) && 
-        (opr2->gtIntCon.gtIconVal == 0))
+        opr2->IsIntegralConst(0))
```

Comment:
There are checks later that the expressions involved are TYP_INT. I don't think it fundamentally should be necessary, but since it's there, I've re-added the check that it's GT_CNS_INT, so that we don't go further down the path before failing to match.